R/zchunk_LA154.Transport_china.R

```diff
@@ -0,0 +1,196 @@
+    L100.Pop_thous_province <- get_data(all_data, "L100.Pop_thous_province")
+    L101.inNBS_Mtce_province_S_F <- get_data(all_data, "L101.inNBS_Mtce_province_S_F")
+    L101.NBS_use_all_Mtce <- get_data(all_data, "L101.NBS_use_all_Mtce")
+    # ===================================================
+
+      # Silence package notes
+      GCAM_region_ID <- year <- value <- UCD_sector <- size.class <- UCD_technology <- UCD_fuel <- fuel <- EBProcess <- EBMaterial <-
+      fuel_sector <- province <- sector <- value_national <- value_share <- pop <- value_mode <- NULL
+
+      # Calculate the province-wise percentages for each of NBS's sector/fuel combinations that is relevant for disaggregating
+      # nation-level transportation energy to the provinces
+
+      # This starting table is transportation energy consumption by GCAM region (and other variables)
+      # We will first subset this data for only the china and values that are > 0 in the historical periods
+
+      L154.in_EJ_R_trn_m_sz_tech_F_Yh %>%
+        filter(year %in% HISTORICAL_YEARS, GCAM_region_ID == 11) %>% # Filter for the china and for historical years only
+        filter(value != 0) %>% # Here any rows with value of 0 will be lost, even if other years of the same group are nonzero
+        # We will next reintroduce those rows using "complete" and assign those values to be 0
+        complete(nesting(GCAM_region_ID, UCD_sector, mode, size.class, UCD_technology, UCD_fuel, fuel), year = HISTORICAL_YEARS, fill = list(value = 0)) %>%
+        # Fuel and mode will be mapped to NBS fuel and sector
+        left_join_error_no_match(trnUCD_NBS_mapping, by = c("fuel", "mode")) ->
+        L154.in_EJ_CHINA_trn_m_sz_tech_F_Yh
+
+
+      #next, extract the relevant NBS sector & fuel combinations from the full province database
+      L101.NBS_use_all_Mtce %>%
+        filter(year %in% HISTORICAL_YEARS) %>% # Ensure within historical period
+        mutate(fuel_sector = paste(EBProcess, EBMaterial)) %>% # Create concatenated list in base dataframe to match the syntax of our list above
+        filter(fuel_sector %in% paste(L154.in_EJ_CHINA_trn_m_sz_tech_F_Yh$EBProcess, L154.in_EJ_CHINA_trn_m_sz_tech_F_Yh$EBMaterial)) %>% # Filtering for just NBS-fuel/sector pairs
+        select(province, EBProcess, EBMaterial, sector, fuel, year, value) %>%
+        group_by(province, year) %>%
+        # First zero out NAs in years where some values are NA but not all
+        mutate(value = replace(value, is.na(value) & sum(value, na.rm = T) != 0, 0)) %>%
+        ungroup %>%
+        # use approx_fun rule = 2 to fill out data in years where the entire province is NA
+        group_by(province, EBProcess, EBMaterial) %>%
+        mutate(value = approx_fun(year, value, rule = 2)) %>%
+        ungroup() ->
+        L154.NBS_trn_Mtce_province
+
+      # To calculate the province share, we need to calculate the national amount
+      L154.NBS_trn_Mtce_province %>%
```

Comment:
You can simply do:
`L154.NBS_trn_Mtce_province %>%`
  `group_by(EBProcess, EBMaterial, year) %>%`
  ` mutate(share = value / sum(value, na.rm = T)) %>%`